models/spring-ai-bedrock-converse/src/main/java/org/springframework/ai/bedrock/converse/BedrockProxyChatModel.java

```diff
@@ -354,6 +361,91 @@ else if (prompt.getOptions() instanceof ChatOptions) {
 			.build();
 	}
 
+	private ContentBlock mapMediaToContentBlock(Media media) {
+
+		var mimeType = media.getMimeType();
```

Comment:
should look to break up this multi-screen function into if/else structure with additional helper methods.